### PR TITLE
Code cleanup

### PR DESCRIPTION
**Pull request by techlover1**
First round of cleaning up the code base, mostly following suggestions from codacy

### CODE REVIEW

File: olly/middleware.py [hunk 1]
```diff
@@ -4,15 +4,27 @@
 from profiles.models import UserProfile, BannedUser
 
 
-class CheckBanListMiddleware:
-    def __init__(self, get_response):
-        self.get_response = get_response
+def tenant_middleware(get_response):
```
Comment (mikemaddem) [inline olly/middleware.py]: Can we get some documentation on this so people know what and how this works?

File: olly/middleware.py [hunk 2]
```diff
@@ -4,15 +4,27 @@
...
+
+        else:
+            request.tenant = 'stock'
+
+        # all done, the view will receive a request with a tenant attribute
+        return get_response(request)
+
+    return middleware
 
-    def __call__(self, request):
```
Comment (mikemaddem) [inline olly/middleware.py]: why is this being removed? django makes a call to this I believe...?

File: olly/middleware.py (hunk 1 above)
Comment (techlover1) [reply olly/middleware.py]: I plan on removing it as it never worked fully as intended. #34 and #10 will work together as an alternative

File: singletournaments/views.py [hunk 3]
```diff
@@ -89,8 +89,8 @@ def post(self, request, pk):
             users = TeamInvite.objects.filter(team=form.data['teams'], accepted=True)
             teams = tournament.teams.all()
             teameligible = False
-            utc = pytz.UTC
-            now = utc.localize(datetime.datetime.now())
+            #utc = pytz.UTC
```
Comment (mikemaddem) [inline singletournaments/views.py]: these were unused variables?

File: staff/views/singletournaments.py [hunk 4]
```diff
@@ -261,11 +250,11 @@ def advance(request, pk):
         matches = currentround.matches.all()
         for i in matches:
             if i.winner is None:
-                if mikes_super_function(tournament.id, currentround.id, nextround.id) is False:
```
Comment (mikemaddem) [inline staff/views/singletournaments.py]: Going to assume you've tested these changes you've made

File: olly/middleware.py (hunk 2 above)
Comment (techlover1) [reply olly/middleware.py]: I have tested it and it works properly without it

File: singletournaments/views.py (hunk 3 above)
Comment (techlover1) [reply singletournaments/views.py]: At this point, yes

File: profiles/views.py [hunk 5]
```diff
@@ -99,7 +99,7 @@ def logout(request, next_page=None,
         next_page = request.POST.get(redirect_field_name,
                                      request.GET.get(redirect_field_name))
         # Security check -- don't allow redirection to a different host.
-        if not is_safe_url(url=next_page, host=request.get_host()):
+        if not is_safe_url(url=next_page, allowed_hosts=request.get_host()):
```
Comment (mikemaddem) [inline profiles/views.py]: what's going on here?

Comment (techlover1) [reply profiles/views.py]: This was a change in Django 2.1

`allowed_hosts is now a required argument of private API django.utils.http.is_safe_url().` and `The host parameter of django.utils.http.is_safe_url() is removed.`

File: staff/views/singletournaments.py (hunk 4 above)
Comment (techlover1) [reply staff/views/singletournaments.py]: This hasn't been tested by hand, however the extra variables are never used in the function

**Comment by techlover1** [comment]
> 
> 
> Would be nice if we added some documentation for the mt middleware.
> Want to ensure these changes, especially to tournaments, have been tested fully.

I plan on the MT middleware as it never worked fully as intended. #34 and #10 will work together as an alternative

**Pull request merged**